src/route/index.ts

```diff
@@ -194,10 +195,13 @@ export async function initApiServer(
         ) => {
           const pubKey = request.params["pubKey"];
           const { network, horizon_url, soroban_url } = request.query;
+
+          const skipSorobanPubnet = network === "PUBLIC" && !useSorobanPublic;
           const contractIds = request.query["contract_ids"] || ([] as string[]);
+
           const { data, error } = await mercuryClient.getAccountBalances(
             pubKey,
-            contractIds,
+            skipSorobanPubnet ? [] : contractIds,
```

Comment:
We can't just return 400 here because this also gets the classic balances so I just pass an empty array for contract IDs to avoid the Soroban path.